### ISSUE TITLE
Transport Entities

### ISSUE THREAD

**Issue by Neuheit** [DSharpPlus/DSharpPlus#571]
# Summary
There are several "transport" entities that can be found [here](https://github.com/DSharpPlus/DSharpPlus/tree/master/DSharpPlus/Net/Abstractions) that seem unnecessary to have.

# Details 
From what I could gather, these transport entities are serialized by Newtonsoft which are then used to initialize the actual entities, either in the client or that entity's constructor. I'm confused on why the library does this, as it would make more sense to just serialize those properties directly (like what is done in `DiscordGuild`).

Here is an example:
```csharp
xp.Activity = new DiscordActivity(xp.RawActivity);
```

This just seems unneeded when the raw activity could have just been the actual activity. Is there a reason why the lib does it this way, and not directly like in a `DiscordGuild`?

**Comment by Emzi0767** [comment]
It seems my comment has vanished, so here it goes again:

tl;dr: Structure is different

**Comment by Neuheit** [comment]
If that is the case then, shouldn't we just use the structure that best reflects the API? In the case of TransportMember, why would that be used instead of just a normal DiscordMember?

**Comment by WamWooWam** [comment]
If I'm frank, I feel we should be using Transport* entities way more than we actually are. Separate the serialisation layer from the actual API surface layer.

**Comment by Emzi0767** [comment]
Yup

**Comment by Neuheit** [comment]
Yeah, make it more consistent.
